src/components/markdown-htmlview.component.js

```diff
@@ -0,0 +1,356 @@
+    rendered = rendered.replace(/(\s)#(\d+)/, (match, spacing, number) => {
+      return `${spacing}<issue src="${number}" />`;
+    });
+    rendered = rendered.replace(
+      /(\s)@([_0-9A-Za-z]+)/,
+      (match, spacing, username) => {
+        return `${spacing}<profile src="${username}" />`;
+      }
+    );
+
+    return rendered;
+  };
+
+  navigateToIssue = number => {
+    const params = this.props.navigation.state.params;
+    const issueURL = params.issue
+      ? `${params.issue.repository_url}/issues/${number}`
+      : params.issueURL.replace(/\d+$/, number);
+
+    this.props.navigation.navigate('Issue', {
+      issueURL,
+    });
+  };
+
+  navigateToProfile = username => {
+    this.props.navigation.navigate('Profile', {
+      user: { login: username },
+    });
+  };
+
+  render() {
+    // console.log(this.props.source);
+    const rendererStyles = {
+      newLine: {
+        // backgroundColor: '#00FF00',
+        height: 5,
+        width: 5,
+      },
+      p: {
+        flexDirection: 'row',
+        flexWrap: 'wrap',
+        margin: 0,
+        padding: 0,
+      },
+      li: {
+        flexDirection: 'row',
+        flexWrap: 'wrap',
+        flexGrow: 1,
+        margin: 0,
+        padding: 0,
+        backgroundColor: '#ffcc00',
+      },
+      h1: {
+        text: {
+          fontSize: normalize(24),
+        },
+        wrapper: {
+          borderBottomWidth: 1,
+          marginBottom: 12,
+        },
+      },
+      h2: {
+        text: {
+          fontSize: normalize(20),
+        },
+        wrapper: {
+          borderBottomWidth: 1,
+          marginBottom: 12,
+        },
+      },
+      h3: {
+        text: {
+          fontSize: normalize(18),
+        },
+      },
+      h4: {
+        text: {
+          fontSize: normalize(16),
+        },
+      },
+      h5: {
+        text: {
+          fontSize: normalize(14),
+        },
+      },
+      h6: {
+        text: {
+          fontSize: normalize(12),
+        },
+      },
+    };
+    const myDomElement = (node, index, siblings, parent, defaultRenderer) => {
+      // const onLinkPress = this.props.onLinkPress;
+
+      /* eslint-disable default-case */
+      switch (node.type) {
+        case 'text':
+          if (node.data === '\n') {
+            return <Text style={rendererStyles.newLine} />;
+          }
+
+          /* eslint-disable no-param-reassign */
+          // Convert checkboxes
+          node.data = node.data.replace(
+            /^\s*\[\s?\]/,
+            emojis.white_large_square
+          );
+          node.data = node.data.replace(/^\s*\[x\]/i, emojis.white_check_mark);
+          // Convert emojis
+          node.data = node.data.replace(/:(\w+):/g, text => {
+            const emoji = text.replace(/:/g, '');
+
+            return emojis[emoji] ? emojis[emoji] : text;
+          });
+
+          break;
+
+        case 'tag':
+          switch (node.name) {
```

Comment:
Personally, I'm not a big fan of `switch` statements (also they add complexity to the codebase).
What do you think about moving this into some kind of function? Or maybe use the strategy pattern to solve this?

I'd love to see a solution like that :)

Reply:
Do you mean something like this? http://robdodson.me/javascript-design-patterns-strategy/ (seems indeed way more readable)

Reply:
Yes!